### PR TITLE
change .then chain to Async/await

### PR DESCRIPTION
**Pull request by OrionStar25**

### CODE REVIEW

File: src/core/issues-notes.js [hunk 1]
```diff
@@ -141,31 +141,40 @@ export function run(conf, doc, cb) {
...
-        }).catch(err => {
-          pub("error", err.message);
-          handleIssues($ins, ghIssues, issueBase);
-          cb();
-        });
-    } else {
+    	async function doAsyncOp () {
+  			try {
+    			await ghFetch(conf.githubAPI);
+    			await fetchJSON(json);
```
Comment (himanish-star) [inline src/core/issues-notes.js]: Where are you getting `json` from?
Do something like` json = await ghFetch(conf.githubAPI);` same for `issues`
_____
Moreover you haven't made a function call of `doAsyncOp()`, so how would the function execute?

File: npm-debug.log [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+4 verbose stack     at FSReqWrap.cb [as oncomplete] (fs.js:212:19)
+5 verbose cwd /home/orionstar/respec
+6 error Linux 4.10.0-30-generic
+7 error argv "/usr/bin/nodejs" "/usr/bin/npm" "run" "babel:" "watch"
+8 error node v4.2.6
+9 error npm  v3.5.2
+10 error missing script: babel:
+11 error If you need help, you may report this error at:
+11 error     <https://github.com/npm/npm/issues>
+12 verbose exit [ 1, true ]
```
Comment (shikhar-scs) [inline npm-debug.log]: This file can be unstaged/removed using a force push. Its not needed here.

File: src/core/issues-notes.js [hunk 3]
```diff
@@ -141,31 +141,47 @@ export function run(conf, doc, cb) {
...
-        });
-    } else {
+    	function fetchIssues(json) {
+			issueBase = issueBase || json.html_url + "/issues/";
+    		return fetchIndex(json.issues_url, {
+        		// Get back HTML content instead of markdown
+        		// See: https://developer.github.com/v3/media/
+        		Accept: "application/vnd.github.v3.html+json",    
+       		});
+		}
```
Comment (shikhar-scs) [inline src/core/issues-notes.js]: nit: indentation.

File: src/core/issues-notes.js [hunk 4]
```diff
@@ -141,31 +141,47 @@ export function run(conf, doc, cb) {
...
+  		doAsyncOp();
+  	} else {
       handleIssues($ins, ghIssues, issueBase);
       cb();
     }
   } else {
     cb();
   }
-}
+};
```
Comment (shikhar-scs) [inline src/core/issues-notes.js]: unnecessary lines follow.

File: npm-debug.log (hunk 2 above)
Comment (OrionStar25) [reply npm-debug.log]: @himanish-star @shikhar-scs thank youboth for your feedback! I was travelling and hence this was done in a hurry but I shall definitely incorporate your changes. Also, my apt-get is malfunctioning :/ Anyway, I'll try to correct it.

**Comment by OrionStar25** [comment]
Another solution for the .then chain as discussed in #1502

File: npm-debug.log (hunk 2 above)
Comment (shikhar-scs) [reply npm-debug.log]: No Problem :sweat_smile:

**Comment by marcoscaceres** [comment]
@OrionStar25, thanks for putting this together. See the feedback in https://github.com/w3c/respec/pull/1533, as the comments I left there apply equally here.

**Comment by marcoscaceres** [comment]
Closing, as this was just an exercise. Great input @OrionStar25. Thanks @shikhar-scs and @himanish-star for your helpful tips here on code quality and spotting a few bugs!

**Pull request closed without merging**